packages/graphql/src/translate/translate-top-level-match.ts

```diff
@@ -20,47 +20,51 @@
 import type { AuthOperations, Context, GraphQLWhereArg } from "../types";
 import type { Node } from "../classes";
 import { createAuthAndParams } from "./create-auth-and-params";
+import { lowerFirst } from "../utils/lower-first";
 import Cypher from "@neo4j/cypher-builder";
 import { createWherePredicate } from "./where/create-where-predicate";
+import { SCORE_FIELD } from "../graphql/directives/fulltext";
 
 export function translateTopLevelMatch({
+    matchNode,
     node,
     context,
     varName,
     operation,
 }: {
+    matchNode?: Cypher.NamedNode;
     context: Context;
     node: Node;
     varName: string;
     operation: AuthOperations;
 }): Cypher.CypherResult {
-    const matchQuery = createMatchClause({ node, context, varName, operation });
+    const matchQuery = createMatchClause({ matchNode, node, context, varName, operation });
 
     const result = matchQuery.build();
     return result;
 }
 
 export function createMatchClause({
+    matchNode,
     node,
     context,
     varName,
     operation,
 }: {
+    matchNode?: Cypher.NamedNode;
     context: Context;
     node: Node;
     varName: string;
     operation: AuthOperations;
 }): Cypher.Match | Cypher.db.FullTextQueryNodes {
     const { resolveTree } = context;
-    const whereInput = resolveTree.args.where as GraphQLWhereArg;
     const fulltextInput = (resolveTree.args.fulltext || {}) as Record<string, { phrase: string }>;
-
-    const matchNode = new Cypher.NamedNode(varName, { labels: node.getLabels(context) });
-
+    matchNode = matchNode || new Cypher.NamedNode(varName, { labels: node.getLabels(context) });
```

Comment:
I don't really like this logic, if both `matchNode` and varName are `defined` now we don't have a source of truth for our name.

Ideally, we wouldn't need to use varName here and we would only use the given node.

If that is not possible, it is probably better to revert to the previous behaviour or accept `varName` as a `Cypher.Node | string`

Reply:
I've refactored this so matchNode is always passed and is the only value that is used (not `varName`)